### PR TITLE
884 data update near surface temperature met office hadley centre

### PR DESCRIPTION
**Pull request by pabloarosado**
I updated this dataset (which didn't really need an update) just to adapt it to the new walkthrough workflow and to see if everything works well. Nothing has changed with respect to the previous steps, so there's no need to review the steps.

Everything went well, except for one small issue: Initially, in the garden metadata, I only had:
```yaml
dataset:
  title: Near-surface temperature anomaly (Met Office Hadley Centre, 2022b)
```
Given that all metadata was well defined since the snapshot step. However, on grapher admin I saw that the datset description (which shows in the SOURCES tab) was duplicated. That's why I had to include `description: ""`. I think this happens because, when transferring the ETL grapher step to grapher, the sources' description are appended to the dataset description.

I think in general it's good to concatenate dataset description and sources' description, except in this kind of scenarios. The only downside is that it creates an extra line at the beginning of the description. @Marigold do you think there's a better solution? Thanks.

### CODE REVIEW

File: etl/steps/data/garden/met_office_hadley_centre/2023-01-17/near_surface_temperature.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+"""Load a meadow dataset and create a garden dataset."""
```
Comment (Marigold) [inline etl/steps/data/garden/met_office_hadley_centre/2023-01-17/near_surface_temperature.py:1]: I'm getting different values in both versions, here's an example

```
from owid.catalog import Dataset
from etl.paths import DATA_DIR
ds = Dataset(DATA_DIR / "garden/met_office_hadley_centre/2023-01-17/near_surface_temperature")
print(ds['near_surface_temperature'].loc['Southern hemisphere'].loc[2022])

ds = Dataset(DATA_DIR / "garden/met_office_hadley_centre/2023-01-02/near_surface_temperature")
print(ds['near_surface_temperature'].loc['Southern hemisphere'].loc[2022])
```

File: etl/steps/data/meadow/met_office_hadley_centre/2023-01-17/near_surface_temperature.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    df = df.set_index(["region", "year"], verify_integrity=True).sort_index()
+
+    # Create a new table and ensure all columns are snake-case.
+    tb = Table(df, short_name=paths.short_name, underscore=True)
+
+    #
+    # Save outputs.
+    #
+    # Create a new meadow dataset with the same metadata as the snapshot.
+    ds_meadow = Dataset.create_empty(dest_dir, metadata=convert_snapshot_metadata(snapshots["Global"].metadata))
```
Comment (Marigold) [inline etl/steps/data/meadow/met_office_hadley_centre/2023-01-17/near_surface_temperature.py:66]: This inherits short name `near_surface_temperature**_global**` from the snapshot. The step itself is called `near_surface_temperature` which causes confusion. I wonder if we should make it mandatory that `namespace/version/short_name` matches path of a dataset.

**Review by Marigold — APPROVED** [review]
Datadiff showed that new version has different values. Would you mind checking it first before merging?

**Comment by Marigold** [comment]
> I think in general it's good to concatenate dataset description and sources' description, except in this kind of scenarios. The only downside is that it creates an extra line at the beginning of the description. @Marigold do you think there's a better solution? Thanks.

For this simple case we can only add the description if it is different from source's description. I'll add that in a different PR. (using `description: ""` is totally fine for this PR)

File: etl/steps/data/garden/met_office_hadley_centre/2023-01-17/near_surface_temperature.py (hunk 1 above)
Comment (pabloarosado) [reply etl/steps/data/garden/met_office_hadley_centre/2023-01-17/near_surface_temperature.py:1]: Thanks for pointing this out, the data for northern and southern hemisphere has changed significantly, I don't know why... I'm checking.

Comment (pabloarosado) [reply etl/steps/data/garden/met_office_hadley_centre/2023-01-17/near_surface_temperature.py:1]: Ouch, it was my mistake. It happened when I manually copied the snapshot metadata of the northern and southern files... I also accidentally copied the URLs. Thanks a lot for pointing this out!

Comment (pabloarosado) [reply etl/steps/data/garden/met_office_hadley_centre/2023-01-17/near_surface_temperature.py:1]: The new data now is identical, except for the latest data point, which has changed a bit. But this is expected.

**Pull request merged**